packages/react/presence/src/Presence.tsx

```diff
@@ -117,6 +117,9 @@ function usePresence(present: boolean) {
         node.removeEventListener('animationcancel', handleAnimationEnd);
         node.removeEventListener('animationend', handleAnimationEnd);
       };
+    } else {
+      // Transition to the unmounted state if the node is removed prematurely
+      send('ANIMATION_END');
```

Comment:
Reading your description of the issue, I was sorta expecting to see:

```jsx
return () => {
  node.removeEventListener('animationstart', handleAnimationStart);
  node.removeEventListener('animationcancel', handleAnimationEnd);
  node.removeEventListener('animationend', handleAnimationEnd);
  // Transition to the unmounted state if the node is removed prematurely
  send('ANIMATION_END');
};
```

Since the cleanup is in a `if (node) {}` check, it will cleanup when `node` becomes `null`. 

Would that work?

Reply:
It seems like it would be equivalent to me but perhaps I'm missing a case?

Reply:
Yeah I started here and could have gone either way, though after working on https://github.com/radix-ui/primitives/pull/1108 I realised they’re not technically equivalent as the node could change or new deps could be introduced in the future.

Reply:
Haha yeah good point, teaches us something 😀

Reply:
Yeah, definitely a good catch. Worth adding a code comment here too about that perhaps?